src/main/java/org/embulk/input/gcs/GcsFileInputPlugin.java

```diff
@@ -64,10 +65,10 @@ public ConfigDiff transaction(final ConfigSource config, final FileInputPlugin.C
             }
         }
 
-        // @see https://cloud.google.com/storage/docs/bucket-naming
+        // @see https://cloud.google.com/storage/docs/objects#naming
         if (task.getLastPath().isPresent()) {
-            if (task.getLastPath().get().length() >= 128) {
-                throw new ConfigException("last_path length is allowed up to 127 characters");
+            if (task.getLastPath().get().getBytes(StandardCharsets.UTF_8).length >= 1025) {
+                throw new ConfigException("last_path length is allowed up to 1024 bytes.");
```

Comment:
Good, but can it be a bit more descriptive?

```suggestion
                throw new ConfigException("last_path is too long, which can contain a maximum of 1024 bytes encoded in UTF-8.");
```